sdk/eventhub/Azure.Messaging.EventHubs/tests/Authorization/SharedAccessSignatureTests.cs

```diff
@@ -56,7 +57,9 @@ public void ToStringReflectsTheValue()
         [TestCase("")]
         public void CompositeConstructorValidatesTheResource(string resource)
         {
-            Assert.That(() => new SharedAccessSignature(resource, "Yay", "OMG!", TimeSpan.FromMilliseconds(500)), Throws.ArgumentException);
+            ExactTypeConstraint typeConstraint = resource is null ? Throws.ArgumentNullException : Throws.ArgumentException;
```

Comment:
*Note to self: *  These feel like a good candidate for an extension method.

Reply:
If you're thinking of extending what `That()` returns, I agree. xUnit - with which I'm more familiar - also has `Assert.Throws<T>(string paramName, Action action) where T : ArgumentException` that allow you to validate a particular parameter name (i.e. make sure you're getting an `ArgumentException` for the right parameter). Such an extension method might also do that.

Reply:
I was thinking more of doing a string extension within the tests, since we're generating that constraint frequently.   Something like:

```csharp
Assert.That() => new Blah(stringValue, stringValue.GetExceptionConstraint());
```

Or something less awkward.   I haven't fully thought it through, but may play with it as time allows.

Reply:
If you weren't concerned with precision, you could have done these as:

```csharp
Assert.That(() => new Blah(stringValue), Throws.InstanceOf<ArgumentException>());
```

You wouldn't be proving ArgumentNull vs Argument, but it's probably good enough outside the `Argument` tests themselves.